freshmaker/pyxis_gql.py

```diff
@@ -496,3 +548,58 @@ def find_images_by_names(self, names):
             page_num += 1
 
         return images
+
+    def find_images_by_repository(self, repo):
+        """Find images which have the provided repository name
+
+        :param string repository: repository name to filter by
+        :return: List of image data
+        :rtype: list
+        """
+        images = []
+
+        query_filter = {}
+        query_filter["and"] = []
+
+        query_filter["and"].append({"repositories": {"repository": {"eq": repo["repository"]}}})
```

Comment:
`repo` is a string by design in your doc string, but here it's used as a dict. I agree the api should be designed to accept a string, use dict as a parameter is unnecessary and making the API hard to use.